modules/imgproc/src/color_lab.cpp

```diff
@@ -2002,6 +2002,10 @@ struct RGB2Lab_f
 
             for(; i < n; i += 3, src += scn)
             {
+                if (cvIsNaN(src[bIdx]) || cvIsNaN(src[1]) || cvIsNaN(src[bIdx^2])) {
```

Comment:
Such explicit extra checks introduce performance degradation which we want to avoid.

Why is previous conversation closed?

Reply:
Sorry I closed because my patch was fairly different.
I reduced the NaN comparison to the bare minimum and added a test.

Reply:
What is about: `cvIsNaN(src[0] + src[1] + src[2])`?

Reply:
Done.
That results in less operations for SIMD.